lib/ws-channel.js

```diff
@@ -554,15 +586,17 @@ WebsocketChannel.prototype._fatalError = function(err) {
     err = new Error(err);
   debug('detach on fatal error: %s', err.message);
   this._detach();
+  this._closeWebsocket();
+  this._closed = err.message;
```

Comment:
I thought about that, but then there wasn't any code that could use the error if it was stored here (and fatal errors are already emitted), so .... YAGNI.